conda_lock/common.py

```diff
@@ -92,25 +90,31 @@ def suffix_union(
     >>> suffix_union([[1], [2, 1], [4, 1]])
     Traceback (most recent call last)
         ...
-    RuntimeError: [4, 1] is not a subset of [2, 1]
+    ValueError: [4, 1] is not an ordered subset of [2, 1]
 
     """
-    from genericpath import commonprefix
+    return list(reversed(prefix_union([list(reversed(s)) for s in collections])))
 
-    result: List["SupportsRichComparisonT"] = []
+
+def prefix_union(collections: Iterable[Sequence]) -> List:
```

Comment:
I wonder if part of the confusion has stemmed from the generic naming of these functions.

Could it be instead called `unify_package_sources` or `merge_package_sources`?

FWIW the way I reason about this logic is a bit different, not sure if this is more or less confusing to others (probably a tad less performant):

```python
def unify_package_sources(collections: Iterable[Sequence[str]]) -> List[str]:
    result = max(collections, key=len)
    for collection in collections:
        if collection != result[-len(collection):]:
            raise ValueError(f"{collection} is not an ordered subset of {result}")
    return result
```

Reply:
Nice!! I think there's an edge case when `len(collection) == 0`. But making sure that's in order, would you like to make a PR to supersede mine?

Reply:
Happy to do so :+1: